src/Orchard.Web/Modules/Orchard.AntiSpam/Drivers/ReCaptchaPartDriver.cs

```diff
@@ -67,67 +70,55 @@ public class ReCaptchaPartDriver : ContentPartDriver<ReCaptchaPart> {
                 return null;
             }
 
-            var submitViewModel = new ReCaptchaPartSubmitViewModel();
-
-            if(updater.TryUpdateModel(submitViewModel, String.Empty, null, null)) {
-                var context = workContext.HttpContext;
-
-                try {
-                    var result = ExecuteValidateRequest(
-                        settings.PrivateKey,
-                        context.Request.ServerVariables["REMOTE_ADDR"],
-                        submitViewModel.recaptcha_challenge_field,
-                        submitViewModel.recaptcha_response_field
-                        );
-
-                    if (!HandleValidateResponse(context, result)) {
-	                _notifier.Error(T("The text you entered in the Captcha field does not match the image"));
-                    	updater.AddModelError("", T("The text you entered in the Captcha field does not match the image"));
+            var context = workContext.HttpContext;
+
+            try {
+                var result = ExecuteValidateRequest(
+                    settings.PrivateKey,
+                    context.Request.ServerVariables["REMOTE_ADDR"],
+                    context.Request.Form["g-recaptcha-response"]
+                    );
+
+                ReCaptchaPartResponseModel responseModel = _jsonConverter.Deserialize<ReCaptchaPartResponseModel>(result);
+
+                if (!responseModel.Success) {

```

Comment:
Did it compile for you when using a string?
In the layout element equivalent you are using a boolean, so I updated this view model to use the same.
It's on the dev branch, perhaps you can pull from that and test if everything still works, particularly this view model binding? Thank you